app/controllers/concerns/user_authorization.rb

```diff
@@ -22,6 +23,13 @@ def authorize_user!
     authorize(@record, policy_class:)
   end
 
+  def authorize_carrier!
```

Comment:
@samnang I think this is better because the previous implementation only signs out the current user. But it could be that more than one user is signed in, the remaining signed in users would have access to the other carrier